Azure/sonic-snmpagent
# [Namespace]: Reduce time taken during get_all from all namespaces

**- What I did**
To support namespaces, we added a Namespace class with functions to support getting data from all databases.
The Namespace dbs_get_all function , checks if the key exists in the database before executing get_all.
This additional check was causing additional time, which was causing agentx socket to disconnect.

**- How I did it**
- Added new Namespace class variables to get the mapping of interface index and  namespace(or database instance).
- Use the new mapping to directly execute get_all from that specific database instance instead of checking if the key exists in the database.

**- How to verify it**
Make sure unit-test passes.
Make sure single-asic platform test cases are passing and no error message is observed in the syslog.
Check snmp output in multi-asic platform.

**- Description for the changelog**
